drivers/sensor/sht4x/sht4x.c

```diff
@@ -0,0 +1,236 @@
+	}
+
+	*rh_sample = sys_get_be16(&rx_buf[3]);
+	if (sht4x_compute_crc(*rh_sample) != rx_buf[5]) {
+		LOG_ERR("Invalid CRC for RH.");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+/* public API for handling the heater */
+int sht4x_fetch_with_heater(const struct device *dev)
+{
+	struct sht4x_data *data = dev->data;
+	int rc;
+
+	rc = sht4x_write_command(dev,
+			heater_cmd[data->heater_power][data->heater_duration]);
+	if (rc) {
+		LOG_ERR("Failed to start measurement.");
+		return rc;
+	}
+
+	k_sleep(K_MSEC(heater_wait_ms[data->heater_duration]));
+
+	rc = sht4x_read_sample(dev, &data->t_sample, &data->rh_sample);
+	if (rc) {
+		LOG_ERR("Failed to fetch data.");
+		return rc;
+	}
+
+	return 0;
+}
+
+static int sht4x_sample_fetch(const struct device *dev,
+			       enum sensor_channel chan)
+{
+	const struct sht4x_config *cfg = dev->config;
+	struct sht4x_data *data = dev->data;
+	int rc;
+
+	if (chan != SENSOR_CHAN_ALL &&
+		chan != SENSOR_CHAN_AMBIENT_TEMP &&
+		chan != SENSOR_CHAN_HUMIDITY) {
+		return -ENOTSUP;
+	}
+
+	rc = sht4x_write_command(dev, measure_cmd[cfg->repeatability]);
+	if (rc) {
+		LOG_ERR("Failed to start measurement.");
+		return rc;
+	}
+	k_sleep(K_USEC(measure_wait_us[cfg->repeatability]));
+
+	rc = sht4x_read_sample(dev, &data->t_sample, &data->rh_sample);
+	if (rc) {
+		LOG_ERR("Failed to fetch data.");
+		return rc;
+	}
+
+	return 0;
+}
+
+static int sht4x_channel_get(const struct device *dev,
+			      enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	const struct sht4x_data *data = dev->data;
+
+	if (chan != SENSOR_CHAN_ALL &&
+		chan != SENSOR_CHAN_AMBIENT_TEMP &&
+		chan != SENSOR_CHAN_HUMIDITY) {
+		return -ENOTSUP;
+	}
+
+	/*
+	 * See datasheet "Conversion of Signal Output" section
+	 * for more details on processing sample data.
+	 */
+	if (chan == SENSOR_CHAN_AMBIENT_TEMP) {
+		int64_t tmp;
+
+		tmp = data->t_sample * 175;
+		val->val1 = (int32_t)(tmp / 0xFFFF) - 45;
+		val->val2 = ((tmp % 0xFFFF) * 1000000) / 0xFFFF;
+	}
+
+	if (chan == SENSOR_CHAN_HUMIDITY) {
+		uint64_t tmp;
+
+		tmp = data->rh_sample * 125U;
+		val->val1 = (uint32_t)(tmp / 0xFFFF) - 6U;
+		val->val2 = (tmp % 0xFFFF) * 15625U / 1024U;
+	}
+
+	return 0;
+}
+
+static int sht4x_attr_set(const struct device *dev,
+				enum sensor_channel chan,
+				enum sensor_attribute attr,
+				const struct sensor_value *val)
+{
+	struct sht4x_data *data = dev->data;
+	enum sensor_attribute_sht4x attr_p = (enum sensor_attribute_sht4x) attr;
+
+	if (attr_p != SENSOR_ATTR_SHT4X_HEATER_POWER &&
+		attr_p != SENSOR_ATTR_SHT4X_HEATER_DURATION) {
+		return -ENOTSUP;
+	}
+
+	if (val->val1 < 0) {
+		return -EINVAL;
+	}
+
+	if (attr_p == SENSOR_ATTR_SHT4X_HEATER_POWER) {
+		if (val->val1 > SHT4X_HEATER_POWER_IDX_MAX) {
+			return -EINVAL;
+		}
+		data->heater_power = val->val1;
+	}
+
+	if (attr_p == SENSOR_ATTR_SHT4X_HEATER_DURATION) {
+		if (val->val1 > SHT4X_HEATER_DURATION_IDX_MAX) {
+			return -EINVAL;
+		}
+		data->heater_duration = val->val1;
+	}
```

Comment:
suggestion: use switch/case or if/else if/else (only a single attribute can be set)

Reply:
updated. much more readable this way